inventory/byo/hosts.origin.example

```diff
@@ -89,6 +89,21 @@ openshift_release=v1.4
 # Skip upgrading Docker during an OpenShift upgrade, leaves the current Docker version alone.
 # docker_upgrade=False
 
+
+# Upgrade Hooks
+#
+# Hooks are available to run custom tasks at various points during a cluster
+# upgrade. Each hook should point to a file with Ansible tasks defined. Suggest using
+# absolute paths, if not the path will be treated as relative to the file where the
+# hook is actually used.
+#
+# Tasks to run before each master is upgraded:
+# openshift_upgrade_pre_master_hook=/usr/share/custom/pre_master.yml
```

Comment:
The hook runs against each master in serial. It can contain only tasks as it needs to plug into the middle of our serial upgrade.

I don't think we can set an interface for what variables are set at this point, the caller is going to have to do their own thing. If they choose to use something of ours, there are presently no guarantees it will continue working going forward.

Hook failure = upgrade failure, you will have to solve and re-run. Hooks will need to be able to accommodate that as upgrade does.

I do not expect hooks to return data, but they can set variables on hosts usable by your later hooks, that seems fine to me.

Does this sound ok, and if so where should I document it? At this point I think we have to keep it simple and expand on things later as need arises. I kind of need to see how it plays out for Ops needs as well. (working on that now)